API/RestSharpAPI/Hooks/RestSharpHooks.cs

```diff
@@ -1,25 +1,109 @@
-﻿using RestSharp;
+﻿using AventStack.ExtentReports;
+using AventStack.ExtentReports.Gherkin.Model;
+using AventStack.ExtentReports.Reporter;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.Configuration;
+using NLog;
+using NLog.Web;
+using NUnit.Framework;
+using RestSharp;
+using System;
+using System.IO;
+using System.Text;
 using TechTalk.SpecFlow;
-using TestFramework.Hooks;
+using TechTalk.SpecFlow.Infrastructure;
+
+using Utilities;
 
 namespace RestSharpApi.Hooks
 {
     [Binding]
-    public sealed class RestSharpHooks 
+    public  class RestSharpHooks 
     {
         public static RestClient _restClient;
+        public static string ProjectPath = Directory.GetParent(Environment.CurrentDirectory).Parent.Parent.FullName;
+        public static string PathReport = ProjectPath + "\\TestResults\\Report\\ExtentReport.html";
+        private static ExtentReports _extent;
+        private static ExtentTest _feature;
+        private static ExtentTest _scenario;
+        public static Logger _logger = LogManager.Setup().LoadConfigurationFromAppSettings().GetCurrentClassLogger();
+        public static EnvironmentConfigSettings config;
+        public IConfiguration Configuration { get; }
+
+        public RestSharpHooks()
+        {
+        }
+
+        [BeforeTestRun]
+        public static void BeforeTestRun()
+        {
+            try
+            {
+                config = TestConfigHelper.GetApplicationConfiguration();
+                Directory.CreateDirectory(ProjectPath + Path.Combine("\\TestResults\\Report"));
```

Comment:
For creating reports, shall we follow the same logic as in UI Tests
1. Store the location of Report folder in appsettings.json
2. Create a folder with the same name as the name of the log file
3. Save the report in logfilenamefolder/reportlocation

Reply:
fixed